sphinx/builders/latex/transforms.py

```diff
@@ -596,3 +596,40 @@ def apply(self):
             section = node.next_node(nodes.section)
             if section:
                 section['ids'].append(':doc')  # special label for :doc:
+
+
+class IndexInSectionTitleTransform(SphinxTransform):
+    """Move index nodes in section title to outside of the title.
+
+    Some kinds of LaTeX macros (cf. fncychap) does not expect index
+    macros are placed inside sectioning node.  This fixes that by
+    moving index nodes to outside of the macros.
```

Comment:
Note: I updated this PR; inserted this docstring comment.

Reply:
perhaps be more precise like

```
   LaTeX index macro is not compatible with some handling of section titles
   such as uppercasing done on LaTeX side (cf. fncychap handling of ``\chapter``).
   Moving the index node to after the title node fixes that.
```

Reply:
updated :-)